server/src/main/java/org/opensearch/index/query/NestedQueryBuilder.java

```diff
@@ -401,16 +401,32 @@ protected void doBuild(SearchContext parentSearchContext, InnerHitsContext inner
                 }
             }
             String name = innerHitBuilder.getName() != null ? innerHitBuilder.getName() : nestedObjectMapper.fullPath();
-            ObjectMapper parentObjectMapper = queryShardContext.nestedScope().nextLevel(nestedObjectMapper);
-            NestedInnerHitSubContext nestedInnerHits = new NestedInnerHitSubContext(
-                name,
-                parentSearchContext,
-                parentObjectMapper,
-                nestedObjectMapper
-            );
-            setupInnerHitsContext(queryShardContext, nestedInnerHits);
-            queryShardContext.nestedScope().previousLevel();
-            innerHitsContext.addInnerHitDefinition(nestedInnerHits);
+            ObjectMapper parentObjectMapper = queryShardContext.nestedScope().getObjectMapper();
+            BitSetProducer parentFilter;
+            if (parentObjectMapper == null) {
+                parentFilter = queryShardContext.getBitsetFilterCache().getBitSetProducer(Queries.newNonNestedFilter());
+            } else {
+                parentFilter = queryShardContext.getBitsetFilterCache().getBitSetProducer(parentObjectMapper.nestedTypeFilter());
```

Comment:
Checking similar logic in `SortBuilder` (https://github.com/opensearch-project/OpenSearch/blob/09bacee5fc85676e97bee6b4ad87dec35c6aa8cc/server/src/main/java/org/opensearch/search/sort/SortBuilder.java#L207-L213), could this just be:

```
 final Query parentQuery; 
 if (parentObjectMapper == null) { 
     parentQuery = Queries.newNonNestedFilter(); 
 } else { 
     parentQuery = parentObjectMapper.nestedTypeFilter(); 
 }
parentFilter = queryShardContext.bitsetFilter(parentQuery);
```

Internally, `queryShardContext` seems to manage the bitset filter cache.

Reply:
Right. This is innerHit query for nested query. Therefore, I just wanted it to be consistence with its nested query builder code.

https://github.com/opensearch-project/OpenSearch/blob/2fd2c3414c27b6f6d501130fd0b5a39d3e5a5139/server/src/main/java/org/opensearch/index/query/NestedQueryBuilder.java#L314-L332

Reply:
Right -- so that doesn't call `getBitsetFilterCache().getBitSetProducer(...)` either. That's the part that seemed inconsistent to me.

Reply:
Updated the code. 
Seem like `context.bitsetFilter()` is actually same as `context.getBitsetFilterCache().getBitSetProducer()` :)

https://github.com/opensearch-project/OpenSearch/blob/2fd2c3414c27b6f6d501130fd0b5a39d3e5a5139/server/src/main/java/org/opensearch/index/query/QueryShardContext.java#L304-L306